src/masternodeman.cpp

```diff
@@ -909,6 +912,8 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
         CMasternodeVerification mnv;
         vRecv >> mnv;
 
+        pfrom->setAskFor.erase(mnv.GetHash());
+
```

Comment:
We relay `mnv` via `inv` in few places and like for all other cases, when the other node has no data yet, it asks us by sending `inv` back https://github.com/dashpay/dash/blob/master/src/net_processing.cpp#L1419 which updates `setAskFor` https://github.com/dashpay/dash/blob/master/src/net.cpp#L2761. So once we finally get data on the other node, we should clear `setAskFor` there. Or am I missing smth?

Reply:
Ahh I see ok this one was a little convoluted and was a harder path to trace but found it's pushing INV here: https://github.com/dashpay/dash/blob/master/src/masternode.h#L405 so that relays PushInventory to all connected nodes which sets up setAskFor.insert later. That is a good catch and wasn't sure of it when I was implementing I did see it but was not sure. Thanks.

Reply:
Do you think you need a  if(!masternodeSync.IsMasternodeListSynced()) return; after line 915 where you added the setAskFor.erase()? I'm not sure if a MN message can be verified if the list isn't synced?

Reply:
Yep, makes sense not to process `mnv` until mn list is synced. Will fix, thanks! 👍